services/data/src/hooks/useDataQuery.ts

```diff
@@ -57,7 +57,7 @@ export const useDataQuery = (query: QueryMap): QueryRenderInput => {
 
         // Cleanup inflight requests
         return abort
-    }, [context, query, refetchCount])
+    }, [context, refetchCount]) // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Query should be mutable or immutable by design?

Instead of a deep equality check, a ref should be used instead.

If mutable it just holds initial value inside and if mutable it updates the ref with actual value as a side effect.